src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -0,0 +1,412 @@
+            Config.AddAuthorityInfo(new AuthorityInfo(AuthorityType.Aad, authorityUri, true, isDefaultAuthority));
+            return (T)this;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="azureCloud"></param>
+        /// <param name="tenantDomainName"></param>
+        /// <param name="isDefaultAuthority"></param>
+        /// <returns></returns>
+        public T WithAadAuthority(
+            AzureCloud azureCloud,
+            string tenantDomainName,
+            bool isDefaultAuthority)
+        {
+            string authorityUri = GetAuthorityUri(azureCloud, AadAuthorityAudience.AzureAdSpecificDirectoryOnly, tenantDomainName);
+            Config.AddAuthorityInfo(new AuthorityInfo(AuthorityType.Aad, authorityUri, true, isDefaultAuthority));
+            return (T)this;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="azureCloud"></param>
+        /// <param name="authorityAudience"></param>
+        /// <param name="isDefaultAuthority"></param>
+        /// <returns></returns>
+        public T WithAadAuthority(AzureCloud azureCloud, AadAuthorityAudience authorityAudience, bool isDefaultAuthority)
+        {
+            string authorityUri = GetAuthorityUri(azureCloud, authorityAudience);
+            Config.AddAuthorityInfo(new AuthorityInfo(AuthorityType.Aad, authorityUri, true, isDefaultAuthority));
+            return (T)this;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="authorityAudience"></param>
+        /// <param name="isDefaultAuthority"></param>
+        /// <returns></returns>
+        public T WithAadAuthority(AadAuthorityAudience authorityAudience, bool isDefaultAuthority)
+        {
+            string authorityUri = GetAuthorityUri(AzureCloud.AzurePublic, authorityAudience);
+            Config.AddAuthorityInfo(new AuthorityInfo(AuthorityType.Aad, authorityUri, true, isDefaultAuthority));
+            return (T)this;
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="authorityUri"></param>
+        /// <param name="isDefaultAuthority"></param>
+        /// <returns></returns>
+        public T WithAadAuthority(string authorityUri, bool isDefaultAuthority)
+        {
+            Config.AddAuthorityInfo(new AuthorityInfo(AuthorityType.Aad, authorityUri, true, isDefaultAuthority));
+            return (T)this;
+        }
+
+        internal static string GetCloudUrl(AzureCloud azureCloud)
+        {
+            switch (azureCloud)
+            {
+            case AzureCloud.AzurePublic:
+                return "https://login.microsoftonline.com";
+            case AzureCloud.AzureChina:
+                return "https://login.chinacloudapi.cn";
+            case AzureCloud.AzureGermany:
+                return "https://login.microsoftonline.de";
+            default:
```

Comment:
>efault: [](start = 13, length = 7)

this is our current trusted host list: 
            "login.windows.net", // Microsoft Azure Worldwide - Used in validation scenarios where host is not this list
            "login.chinacloudapi.cn", // Microsoft Azure China
            "login.microsoftonline.de", // Microsoft Azure Blackforest
            "login-us.microsoftonline.com", // Microsoft Azure US Government - Legacy
            "login.microsoftonline.us", // Microsoft Azure US Government
            "login.microsoftonline.com", // Microsoft Azure Worldwide
            "login.cloudgovapi.us" // Microsoft Azure US Government

Reply:
Thanks Jenny.  I'll add this to my agenda list for today's meeting.  It looks like a few of these are duplicates of the same logical endpoint (e.g. azure us governemnt, azure worldwide) and the intention here is to have one url as the main entry point to each of the defined clouds.  So i think with adding government we're good but let's discuss.

---
In reply to: [246092090](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/761#discussion_r246092090) [](ancestors = 246092090)